Feature/PropertyDescriptionView implementation

caioalcn opened the pull request:
# Feature

- PropertyDescriptionView implementation;

# Checklist
- [x]  Implementar componente em um arquivo chamado PropertyDescriptionView.swift
- [x]  O componente deve mostrar o título "Descrição”
- [x]  O texto deve ser limitado a 5 linhas
- [x]  Botão deve mostrar o título "Ver mais”
- [x]  O componente deve ser mostrado na Tela de Detalhes do Anúncio

Commit:
Feature:
- Implementar PropertyDescriptionView;

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/PropertyDetails/PropertyDetailsView.swift:
```diff
@@ -0,0 +1,366 @@
+    func setupSubviews() {
+        self.addSubview(scrollView)
+        scrollView.addSubview(contentView)
+        contentView.addSubview(propertyImageView)
```
Que tal utilizarmos  [propertyImageView, valueLabel, condLabel....].forEach {contentView.addSubView($0)}

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/PropertyDetails/PropertyDetailsView.swift:
```diff
@@ -0,0 +1,366 @@
+    private let valueLabel: UILabel = {
+        let label = UILabel()
+        label.text = "R$ 405.000"
+        label.font = UIFont.boldSystemFont(ofSize: 18)
```
Montar uma forma de obter UIFont e Cores em tokens.
18 seria um token
16 outro token.

Por exemplo: 

DesignSystem.typography.xm pode ser o 18
DesignSystem.typography.xs pode ser o 16

O retorno seria diretamente um objeto do UIFont  criado

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/PropertyDetails/PropertyDetailsView.swift:
```diff
@@ -0,0 +1,366 @@
+    private let descriptionLabel: UILabel = {
+        let label = UILabel()
+        label.numberOfLines = 0
+        label.text = " Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec id elit sit amet mauris bibendum feugiat. Aenean blandit ut diam id placerat. Quisque aliquet luctus lacus quis rhoncus. Proin id scelerisque diam. Aenean vehicula vel ipsum in vestibulum. Integer aliquet sollicitudin est. Donec commodo lobortis nulla, sed lobortis sapien efficitur ut. Phasellus vitae condimentu ...
```
Utilizar Localizable key/value, onde value seria a string solta.
Obter esse valor aqui e encapsular as strings do projeto neste localizable.

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/PropertyDetails/PropertyDetailsView.swift:
```diff
@@ -0,0 +1,366 @@
+    }
+    
+    func setupConstraints() {
+        constraintDescriptionHeightMin =             descriptionLabel.heightAnchor.constraint(equalToConstant: 200)
```
Ajustar espaco

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/PropertyDetails/PropertyDetailsView.swift:
```diff
@@ -0,0 +1,366 @@
+        parkLabel.setContentHuggingPriority(UILayoutPriority.fittingSizeLevel, for: .horizontal)
+
+        NSLayoutConstraint.activate([
+            scrollView.safeAreaLayoutGuide.topAnchor.constraint(equalTo: self.safeAreaLayoutGuide.topAnchor),
```
Tentar utilizar alinhamentos com centerY E Menor ou igual na label mais a direita com relacao ao right da view superior

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/PropertyDetails/PropertyDetailsView.swift:
```diff
@@ -0,0 +1,366 @@
+        setupSubviews()
+    }
+    
+    required init?(coder: NSCoder) {
```
Criar uma BaseView de forma que iniba as filhas a implementar init.

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Extensions/UIViewController+Extensions.swift:
```diff
@@ -36,5 +36,14 @@ extension UIViewController {
     }
+    
+    func hideKeyboardWhenTappedAround() {
+        let tapGesture = UITapGestureRecognizer(target: self,
```
Criar uma extension que adicione metodo addTap em qualquer UIView, encapsulando em todas a adicao do gesture

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/PropertyDetails/PropertyDetailsViewController.swift:
```diff
@@ -8,10 +8,54 @@
+    }
+    
+    @objc private func keyboardWillShow(notification: NSNotification) {
+        if let keyboardSize = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue {
```
Infringimos aqui a Lei de Demeter.
Pesquisar sobre e efetuar a correcao :)

andrevinic replied:
if let userInfo = notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue,
let keyboardSize = userInfo.cgRectValue {.....}

Commit:
Fix:
- Change how to add views;
- Change UIFont and Colors using a token;
- Change the strings to a Localizable file;
- Removed spaces left;
- Removed hugging constraint and change to lessOrGreaterThan;
- Created a Base View;
- Created extension to add gesture to UIView;

Commit:
Fix:
- Change the init from the BaseView;
- Change to access property by local variables so we don't break the law of demeter;

andrevinic committed:
Merge branch 'main' into propertydescription-view-interface
